scripts/getChangelog.py

```diff
@@ -1,90 +1,587 @@
+      repository(owner: $owner, name: $repo) {
+        latestRelease {
+          tagName
+        }
+      }
+    }
+    '''
+    while True:
+      try:
+        variables = {"owner": owner, "repo": repo}
+        response = client.execute(query, variables)
+        data = json.loads(response)
+        base_ref = data['data']['repository']['latestRelease']['tagName']
+        print("\nOkido, let's get the latest release (" + base_ref + ")!")
+        break
+      except Exception as e:
+        handle_exception(e, data)
+        print("Does your token have enough permissions?")
+        sys.exit(1)
+  else:
+    query = '''
+    query ($owner: String!, $repo: String!, $ref: String!) {
+      repository(owner: $owner, name: $repo) {
+        ref(qualifiedName:"$ref") {
+          name
+        }
+      }
+    }
+    '''
+    while True:
+      try:
+        variables = {"owner": owner, "repo": repo, "ref": base_ref}
+        response = client.execute(query, variables)
+        data = json.loads(response)
+        if data['data']['repository']['ref'] == None:
+          print("\nError: Invalid release code or branch.")
+          print("Please try again.")
+          base_ref = input('Base: ')
+          continue
+        base_ref = data['data']['repository']['ref']['name']
+        print(f"\nNoted, {base_ref} it is.")
+        break
+      except Exception as e:
+        handle_exception(e, data)
+        print("Does your token have enough permissions?")
+        sys.exit(1)
+
+  # Define head_ref
+  if not head_ref:
+    print("\nThen, from which environment would you like to see the changes (the head)?")
+    print("Please enter a release code or branch")
+    print("[default: 'master']")
+    head_ref = input('Head: ')
+
+  if head_ref == "master" or not head_ref:
+      print("\nRighty, master it is!")
+  if not head_ref:
+    head_ref = "master"
+  else:
+    query = '''
+    query ($owner: String!, $repo: String!, $ref: String!) {
+      repository(owner: $owner, name: $repo) {
+        ref(qualifiedName: $ref) {
+          id
+          name
+        }
+      }
+    }
+    '''
+    while True:
+      try:
+        variables = {"owner": owner, "repo": repo, "ref": head_ref}
+        response = client.execute(query, variables)
+        data = json.loads(response)
+        if data['data']['repository']['ref'] == None:
+          print("\nError: Invalid release code or branch.")
+          print("Please try again.")
+          head_ref = input('Head: ')
+          continue
+        head_ref = data['data']['repository']['ref']['name']
+        print(f"\nNoted, {head_ref} it is.")
+        break
+      except Exception as e:
+          handle_exception(e, data)
+          print("Does your token have enough permissions?")
+          sys.exit(1)
+
+  # Start the animation in a separate thread
+  animation_thread = threading.Thread(target=display_processing_animation, args=(animation_state,))
+  animation_thread.start()
```

Comment:
Do we really need an animation? This just adds complexity to an already quite complex script. Instead, I would just print something like the number of PRs already fetched in the loop. Then one can see that it actually makes progress instead of just looking at a fancy (but not very helpful) animation